network-libp2p/src/network.rs

```diff
@@ -74,8 +75,12 @@ impl Network {
     /// # Arguments
     ///
     ///  - `config`: The network configuration, containing key pair, and other behavior-specific configuration.
+    ///  - `dht_verifier`: The verifier used to verify all Dht records.
     ///
-    pub async fn new(config: Config) -> Self {
+    pub async fn new(
+        config: Config,
+        #[cfg(feature = "kad")] dht_verifier: impl dht::Verifier + 'static,
```

Comment:
This is problematic, the cargo feature `kad` is no longer purely additive. It should probably always take an `Option<impl …>`.